packages/preact/inject/src/init/autocomplete/createElements.tsx

```diff
@@ -0,0 +1,89 @@
+import { CssSelector, resolveCssSelector } from "@preact/inject/resolveCssSelector"
+
+export function createElements(input: HTMLInputElement, dropdown: HTMLDivElement, dropdownSelector: CssSelector) {
+  dropdown.style.display = "none"
+
+  const wrapper = (() => {
+    if (input.parentElement && input.parentElement.classList.contains("nosto-autocomplete-wrapper")) {
+      return input.parentElement
+    }
+
+    const wrapper = document.createElement("div")
+    wrapper.className = "nosto-autocomplete-wrapper"
+    input.parentNode?.insertBefore(wrapper, input.nextSibling)
+    wrapper.appendChild(input)
+    return wrapper
+  })()
+
+  const dropdownLocator = resolveCssSelector(dropdownSelector)
+  const dropdownParent = dropdownLocator && document.querySelector(dropdownLocator.selector)
+
+  if (dropdownParent) {
+    const form = document.createElement("form")
+    form.className = "nosto-dropdown-form"
+    form.appendChild(dropdown)
+    if (dropdownLocator.position === "first") {
+      dropdownParent.prepend(form)
+    } else {
+      dropdownParent.appendChild(form)
+    }
+  } else {
+    wrapper.appendChild(dropdown)
+  }
+
+  const highlightElement = (elementList: Element[], highlightIndex: number) => {
+    dropdown
+      .querySelector(".ns-autocomplete-element.ns-autocomplete-element-hovered")
+      ?.classList.remove("ns-autocomplete-element-hovered")
+    elementList[highlightIndex]?.classList.add("ns-autocomplete-element-hovered")
+  }
+
+  const getHighlightIndex = (removeHighlight?: boolean): number => {
+    const elements = Array.from(dropdown.getElementsByClassName("ns-autocomplete-element"))
+
+    const highlighted = elements.find(item => {
+      if (item.classList.contains("ns-autocomplete-element-hovered")) {
+        if (removeHighlight) {
+          item.classList.remove("ns-autocomplete-element-hovered")
+        }
+        return true
+      }
+      return false
+    })
+
+    return highlighted ? elements.indexOf(highlighted) : -1
+  }
+  return {
+    element: dropdown,
+    hide: () => {
+      dropdown.style.display = "none"
+    },
+    show: () => {
+      dropdown.style.display = ""
```

Comment:
Maybe we should rethink this? In case if the parent is using flexbox, setting display to "" could break the styling. Just a thought. Either use "block" or "inherit"?